apps/site/lib/site_web/controllers/trip_plan_controller.ex

```diff
@@ -20,14 +20,42 @@ defmodule SiteWeb.TripPlanController do
   @type route_map :: %{optional(Route.id_t()) => Route.t()}
   @type route_mapper :: (Route.id_t() -> Route.t() | nil)
 
+  @plan_datetime_selector_fields %{
+    depart: "depart",
+    leaveNow: "leave-now",
+    arrive: "arrive",
+    controls: "trip-plan-datepicker",
+    year: "plan_date_time_year",
+    month: "plan_date_time_month",
+    day: "plan_date_time_day",
+    hour: "plan_date_time_hour",
+    minute: "plan_date_time_minute",
+    amPm: "plan_date_time_am_pm",
+    dateEl: %{
+      container: "plan-date",
+      input: "plan-date-input",
+      select: "plan-date-select",
+      label: "plan-date-label"
+    },
+    timeEl: %{
+      container: "plan-time",
+      select: "plan-time-select",
+      label: "plan-time-label"
+    },
+    title: "trip-plan-departure-title"
+  }
+
   def index(conn, %{"plan" => %{"to" => _to, "from" => _fr} = plan}) do
     conn
     |> assign(:expanded, conn.query_params["expanded"])
+    |> assign(:plan_datetime_selector_fields, @plan_datetime_selector_fields)
```

Comment:
These `assign()` calls can probably be replaced with a plug...see the `optimize_for` function (which is a plug) in this module for an example. Then you could add `plug(:your_plug_function)` at top of this file and it'll run for each of these `index()` calls to assign the variable.

Reply:
Should look better now!

Reply:
Looks great! 👍